### ISSUE TITLE
Follow-up improvements to RAFT training reference

### ISSUE THREAD

**Issue by NicolasHug** [pytorch/vision#5056]
This issue is to keep track of the progress on some minor improvements related to the RAFT training reference, as a follow up to https://github.com/pytorch/vision/issues/4644

- [x] support `--weights` param to support the prototype API, once pre-trained weights are available
- [x] support `--model` instead of the `--small` flag
- [x] support `--epochs` param instead of `--num-steps`, for consistency with the rest of the training references.
- [ ] save the scheduler and the optimizer, not just the model.
- [ ] try to keep function names as close as possible to existing references. For example `validate()` -> `evaluate()`
- [ ] Support `--device` (?? - not sure how useful it would be)
- [ ] Better support of weight serialization (move on CPU, avoid having the DDP model weights etc)

cc @datumbox

**Comment by NicolasHug** [comment]
I beleieve all the remaining points have been addressed by @YosuaMichael in https://github.com/pytorch/vision/pull/5590. Thanks Yosua!
